### PR TITLE
ACI Classpath loading

### PR DESCRIPTION
**Pull request by tbekas**

### CODE REVIEW

File: modules/core/src/main/scala/org/tesselation/domain/aci/StateChannelContext.scala [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package org.tesselation.domain.aci
+
+import org.tesselation.schema.address.Address
+
+trait StateChannelContext[F[_]] {
+  val address: Address
+  def createCell(input: Array[Byte]): F[StdCell[F]]
+}
```
Comment (marcinwadon) [inline modules/core/src/main/scala/org/tesselation/domain/aci/StateChannelContext.scala:8]: IMO it shouldn't be in the `core` module. probably `kernel` is a better place for this
@kpudlik @TheMMaciek ?

File: modules/core/src/main/scala/org/tesselation/infrastructure/aci/StateChannelInputPipeline.scala [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+import cats.syntax.flatMap._
+
+class StateChannelInputPipeline[F[_]: Async](
+  stateChannelInputQueue: Queue[F, StateChannelInput],
+  stateChannelContexts: Map[Address, StateChannelContext[F]]
+) {
+  val logger = Slf4jLogger.getLogger[F]
+
+  Stream.fromQueueUnterminated(stateChannelInputQueue)
```
Comment (marcinwadon) [inline modules/core/src/main/scala/org/tesselation/infrastructure/aci/StateChannelInputPipeline.scala:19]: ?

File: modules/core/src/main/scala/org/tesselation/domain/aci/StateChannelContext.scala (hunk 1 above)
Comment (tbekas) [reply modules/core/src/main/scala/org/tesselation/domain/aci/StateChannelContext.scala:8]: I think that the `kernel` module should contain only the stuff necessary for the users to create the StateChannel. This context is an internal thing, that gets created when loading a jar.

File: modules/core/src/main/scala/org/tesselation/infrastructure/aci/StateChannelInputPipeline.scala (hunk 2 above)
Comment (tbekas) [reply modules/core/src/main/scala/org/tesselation/infrastructure/aci/StateChannelInputPipeline.scala:19]: Yep, I'm addressing it :)

File: modules/core/src/main/scala/org/tesselation/domain/aci/StateChannelContext.scala (hunk 1 above)
Comment (marcinwadon) [reply modules/core/src/main/scala/org/tesselation/domain/aci/StateChannelContext.scala:8]: but related to state channels 🤔 let's discuss it on slack

**Review by marcinwadon — APPROVED** [review]

**Pull request merged**